controllers/chat/initChatFromPost.js

```diff
@@ -114,9 +114,12 @@ const initChatFromPost = async (req, res) => {
     try {
       if (!newChannel?.data?.name) {
         let channelType = CHANNEL_TYPE.CHAT;
-        let anonDetailInfo = {};
-        if (userModel.is_anonymous || targetUserModel.is_anonymous) {
+        if (userModel.is_anonymous) {
           channelType = CHANNEL_TYPE.ANONYMOUS;
+        }
+
+        let anonDetailInfo = {};
+        if (targetUserModel.is_anonymous) {
           anonDetailInfo = {
             anon_user_info_color_code:
               newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code || null,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [122-126]

The update to `anonDetailInfo` only when `targetUserModel.is_anonymous` is true also seems reasonable. But there might be a potential issue here. If `newStateMemberWithAnonInfo[targetUserModel.user_id]` is undefined, trying to access `anon_user_info_color_code` will throw an error. To avoid this, you should add a check before accessing this property.
```diff
- anon_user_info_color_code: newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code || null,
+ anon_user_info_color_code: newStateMemberWithAnonInfo[targetUserModel.user_id]?.anon_user_info_color_code || null,
```
This way, if `newStateMemberWithAnonInfo[targetUserModel.user_id]` is undefined, it won't throw an error and will instead use the fallback value of `null`.


<!-- This is an auto-generated comment by OSS CodeRabbit -->